dcpy/library/validator.py

```diff
@@ -38,37 +38,35 @@ class Socrata(BaseModel):
 
 
 class SourceSection(BaseModel):
-    url: Optional[Url] = None  # Pass another schema as a data type
-    socrata: Optional[Socrata] = None
-    script: Optional[str] = None
-    geometry: GeometryType
-    options: List[str] = []  # Use List[dtype] for a list field value
+    url: Url | None = None
+    script: dict[str, str] | None = None
+    socrata: Socrata | None = None
+    geometry: GeometryType | None = None
+    options: list[str] | None = None
+    gdalpath: str | None = None
 
 
 class DestinationSection(BaseModel):
-    name: str
     geometry: GeometryType
-    options: List[str] = []
-    fields: List[str] = []
-    sql: Optional[str] = None
+    options: list[str] | None = None
+    fields: list[str] | None = None
+    sql: str | None = None
 
 
 class InfoSection(BaseModel):
-    info: Optional[str] = None
-    url: Optional[str] = None
-    dependents: Optional[List[str]] = None
-
-    class Config:
-        extra = Extra.allow
+    info: str | None = None
+    url: str | None = None
+    dependents: List[str] | None = None
 
 
 class Dataset(BaseModel):
```

Comment:
Should we add `extra="forbid"` to the class, to disallow extra fields in the source yml?